src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+enum Shape {
+  Circle = 'circle',
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+}
+
 export interface Figure {
+  color: string,
+  shape: Shape,
+  getArea: () => number,
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+function isValid(...args: number[]): void {
+  args.forEach((arg) => {
+    if (arg <= 0) {
+      throw new Error('Invalid input: some argument <= 0');
+    }
+  });
+}
+
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    isValid(a, b, c);
 
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Invalid input: hypotenuse >= legs sum');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const area: number = 0.25 * Math.sqrt(((a * a + b * b + c * c) ** 2)
```

Comment:
`* 0.25` === ` / 4`
`a * a` === `a ** 2`
`(a * a + b * b + c * c) ** 2` === `a ** 4 + ...` (and it's repeated twice).

Reply:
on my opninion, (a * a + b * b + c * c) ** 2 better leave how it was, because when we assign it, it looks much more complicated
![image](https://user-images.githubusercontent.com/122156399/223737616-cdd2a462-aba2-4b33-b460-b47e2a6d9173.png)